dev/scripts/main.js

```diff
@@ -149,12 +149,16 @@ bikeApp.compareDistances = function() {
 			}
 		});
 
-		const shortestDistanceOriginStation = shortestDistanceOrigin[0];
-		const shortestDistanceDestinationStation = shortestDistanceDestination[0];
-		const shortestDistanceOriginLatLong = new google.maps.LatLng(shortestDistanceOriginStation.station_latlong[0], shortestDistanceOriginStation.station_latlong[1]);
-		const shortestDistanceDestinationLatLong = new google.maps.LatLng(shortestDistanceDestinationStation.station_latlong[0], shortestDistanceDestinationStation.station_latlong[1]);
+		bikeApp.shortestDistanceOriginStation = shortestDistanceOrigin[0];
+		bikeApp.shortestDistanceDestinationStation = shortestDistanceDestination[0];
+		const shortestDistanceOriginLatLong = new google.maps.LatLng(bikeApp.shortestDistanceOriginStation.station_latlong[0], bikeApp.shortestDistanceOriginStation.station_latlong[1]);
+		const shortestDistanceDestinationLatLong = new google.maps.LatLng(bikeApp.shortestDistanceDestinationStation.station_latlong[0], bikeApp.shortestDistanceDestinationStation.station_latlong[1]);
+		//data to be used if there are no bike stations near user destination point
+		bikeApp.distanceBetweenOriginStationAndDestination = google.maps.geometry.spherical.computeDistanceBetween(shortestDistanceOriginLatLong, destinationLatLong);
+		bikeApp.distanceBetweenOriginStationAndDestination = (bikeApp.distanceBetweenOriginStationAndDestination / 1000).toFixed(2);
 		// need latitude and longitude value for shortest distance station to plug into this function
 		bikeApp.getDistanceDuration(shortestDistanceOriginLatLong, shortestDistanceDestinationLatLong);
+		bikeApp.getDistanceDurationRoundTrip(shortestDistanceOriginLatLong, destinationLatLong);
```

Comment:
only call this function after attaching promise to first google api matrix call - move this